### PR TITLE
SP2-4029 first version - git message checker

### PR DESCRIPTION
**Pull request by CoburnJoe**
First draft at the commit message checker script. Still to do:
- Update the regex to work with bitbucket merge commit messages
- Better docs - examples for use with both Bitbucket and GitHub builds

### CODE REVIEW

File: tools/commit-message-checker.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+# Arg checking
+[ -z "$source_branch" ] && echo "Missing source branch" && exit 1
+[ -z "$destination_branch" ] && echo "Missing destination branch" && exit 1
+
+# Message comprehension
+commit_messages=$(git log --pretty=format:%s $destination_branch..$source_branch)
+
+echo "$COMMITS" | while read line ; do
+	if echo "$line" | grep -v -q -E "$PATTERN"; then
```
Comment (nomad-software) [inline tools/commit-message-checker.sh:28]: I'm not sure it matters but an uppercase `$PATTERN` is used here but all other usages are lowercase.

File: tools/commit-message-checker.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+echo "Pattern in use: $pattern"
+
+# Arg checking
+[ -z "$source_branch" ] && echo "Missing source branch" && exit 1
+[ -z "$destination_branch" ] && echo "Missing destination branch" && exit 1
+
+# Message comprehension
+commit_messages=$(git log --pretty=format:%s $destination_branch..$source_branch)
+
+echo "$COMMITS" | while read line ; do
```
Comment (nomad-software) [inline tools/commit-message-checker.sh:27]: Where does `$COMMITS` come from?

File: tools/commit-message-checker.sh [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	if echo "$line" | grep -v -q -E "$PATTERN"; then
+		echo "--- ERROR -- Commit message does not match criteria: $line"
+		# Defer termination to allow every line to spit out the message
+		FAILURE=true
```
Comment (nomad-software) [inline tools/commit-message-checker.sh:31]: Just an idea, `FAILURE` could instead be `$EXIT_CODE`.

**Review by nomad-software — CHANGES_REQUESTED** [review]

File: tools/commit-message-checker.sh (hunk 2 above)
Comment (CoburnJoe) [reply tools/commit-message-checker.sh]: Fixed 👍

File: tools/commit-message-checker.sh (hunk 1 above)
Comment (CoburnJoe) [reply tools/commit-message-checker.sh]: Fixed 👍

File: tools/commit-message-checker.sh (hunk 3 above)
Comment (CoburnJoe) [reply tools/commit-message-checker.sh]: Changed to `requires_exit` 👍

**Review by Marcus-UK — APPROVED** [review]

**Review by nomad-software — APPROVED** [review]

**Review by nomad-software — APPROVED** [review]

**Review by Marcus-UK — APPROVED** [review]

**Pull request merged**